torch/_dynamo/utils.py

```diff
@@ -86,20 +86,86 @@ def profile_wrapper(*args, **kwargs):
     return profile_wrapper
 
 
-def dynamo_timed(func):
-    @wraps(func)
-    def time_wrapper(*args, **kwargs):
-        key = func.__qualname__
-        if key not in compilation_metrics:
-            compilation_metrics[key] = []
-        t0 = time.time()
-        r = func(*args, **kwargs)
-        latency = time.time() - t0
-        # print(f"Dynamo timer: key={key}, latency={latency:.2f} sec")
-        compilation_metrics[key].append(latency)
-        return r
-
-    return time_wrapper
+frame_phase_timing = collections.OrderedDict()
+
+curr_frame = 0
+
+# Note: Called for you by dynamo - you almost never ever want to invoke this yourself.
+def increment_frame():
+    global curr_frame
+    curr_frame = curr_frame + 1
+
+
+# Note: Called for you by dynamo - you almost never ever want to invoke this yourself.
+def reset_frame_count():
+    global curr_frame
+    frame_phase_timing.clear()
+    curr_frame = 0
+
+
+# Print a report of time spent so far
+# Ex:
+# TIMING:
+# entire_frame_compile:8.574629999999999
+# backend_compile:5.26806
+def print_time_report():
+    total = 0
+    total_by_key = {}
+    print("TIMING:")
+    for frame, timings in frame_phase_timing.items():
+        for key, timing in timings.items():
+            total += timing
+            if key not in total_by_key:
+                total_by_key[key] = timing
+            else:
+                total_by_key[key] += timing
+    for key, value in total_by_key.items():
+        print(f"{key}:{round(value, 5)}")
```

Comment:
My preference is to have this as a single line, to make it easier to grep for by automated tools

Reply:
sure